### DISCUSSION TITLE
How can generate an API to acces FH without going through the SDKs

### DISCUSSION THREAD

**Discussion by fastboyz** [featurehub-io/featurehub#418] [Q&A] [answered]
Like the title says, I would like to access the API

**Comment by rvowles** [comment, marked as answer]
Probably the easiest way to copy how to do this is here: 

https://github.com/featurehub-io/featurehub/blob/master/admin-frontend/app_mr_layer/pom.xml

The OpenAPI components are fragmented because they are used internally throughout the infrastructure to communicate between MR, the front end, Dacha and the Edge. The first part of this combines all the pieces that are exposes via the Management Repository and the second one uses a third party Dart plugin (in this case) to generate the API source code. Typically you would not use the extra dependency in the OpenAPI definition and choose one of the "standard" ones.

An example of that can be seen here (for C#, for Edge)

https://github.com/featurehub-io/featurehub/blob/master/sdks/client-csharp/FeatureHubSDK/pom.xml

and here for the Typescript Client SDK (which clearly needs an update!)

https://github.com/featurehub-io/featurehub/blob/master/sdks/typescript/featurehub-javascript-client-sdk/pom.xml
